lib/models/os.ts

```diff
@@ -0,0 +1,676 @@
+	 * @name isSupportedOsUpdate
+	 * @public
+	 * @function
+	 * @memberof balena.models.os
+	 *
+	 * @param {String} deviceType - device type slug
+	 * @param {String} currentVersion - semver-compatible version for the starting OS version
+	 * @param {String} targetVersion - semver-compatible version for the target OS version
+	 * @fulfil {Boolean} - whether upgrading the OS to the target version is supported
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.os.isSupportedOsUpgrade('raspberry-pi', '2.9.6+rev2.prod', '2.29.2+rev1.prod').then(function(isSupported) {
+	 * 	console.log(isSupported);
+	 * });
+	 *
+	 * balena.models.os.isSupportedOsUpgrade('raspberry-pi', '2.9.6+rev2.prod', '2.29.2+rev1.prod', function(error, config) {
+	 * 	if (error) throw error;
+	 * 	console.log(isSupported);
+	 * });
+	 */
+	const isSupportedOsUpdate = (
+		deviceType: string,
+		currentVersion: string,
+		targetVersion: string,
+		callback?: (error?: Error, result?: boolean) => void,
+	): Promise<boolean> =>
+		getValidatedDeviceType(deviceType)
+			.then(() =>
+				hupActionHelper.isSupportedOsUpdate(
+					deviceType,
+					currentVersion,
+					targetVersion,
+				),
+			)
+			.asCallback(callback);
+
+	/**
+	 * @summary Returns the supported OS update targets for the provided device type
+	 * @name getSupportedOsUpdateVersions
+	 * @public
+	 * @function
+	 * @memberof balena.models.os
+	 *
+	 * @param {String} deviceType - device type slug
+	 * @param {String} currentVersion - semver-compatible version for the starting OS version
+	 * @fulfil {Object} - the versions information, of the following structure:
+	 * * versions - an array of strings,
+	 * containing exact version numbers that OS update is supported
+	 * * recommended - the recommended version, i.e. the most recent version
+	 * that is _not_ pre-release, can be `null`
+	 * * current - the provided current version after normalization
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.os.getSupportedOsUpdateVersions('raspberry-pi', '2.9.6+rev2.prod').then(function(isSupported) {
+	 * 	console.log(isSupported);
+	 * });
+	 *
+	 * balena.models.os.getSupportedOsUpdateVersions('raspberry-pi', '2.9.6+rev2.prod', function(error, config) {
+	 * 	if (error) throw error;
+	 * 	console.log(isSupported);
+	 * });
+	 */
+	const getSupportedOsUpdateVersions = (
+		deviceType: string,
+		currentVersion: string,
+		callback?: (error?: Error, result?: OsUpdateVersions) => void,
+	): Promise<OsUpdateVersions> =>
+		getSupportedVersions(deviceType)
+			.then(({ versions: allVersions }) => {
+				// use bSemver.compare to find the current version in the OS list
+				// to benefit from the baked-in normalization
+				const current = find(
+					allVersions,
+					v => bSemver.compare(v, currentVersion) === 0,
+				);
+
+				const versions = allVersions.filter(v =>
+					// avoid the extra call to getValidatedDeviceType, since getSupportedVersions is already does that
```

Comment:
It seems that I had a typo here :sweat_smile: 
```suggestion
					// avoid the extra call to getValidatedDeviceType, since getSupportedVersions already does that
```